operator-pipeline-images/operatorcert/github.py

```diff
@@ -22,18 +24,62 @@ def _get_session(auth_required: bool = False) -> requests.Session:
     Returns:
         requests.Session: Github session
     """
-    token = os.environ.get("GITHUB_TOKEN")
-
-    if auth_required and not token:
-        raise Exception("No auth details provided for Github. Define GITHUB_TOKEN.")
-
     session = requests.Session()
-    session.headers.update(
-        {"Authorization": f"Bearer {token}", "Accept": "application/vnd.github.v3+json"}
+
+    # Exponential retry backoff for a max wait of ~8.5 mins
+    retries = Retry(
+        total=10, backoff_factor=1, status_forcelist=(408, 500, 502, 503, 504)
     )
+    adapter = HTTPAdapter(max_retries=retries)
+    session.mount("http://", adapter)
+    session.mount("https://", adapter)
+
+    if auth_required:
+        token = os.environ.get("GITHUB_TOKEN")
+
+        if not token:
+            raise Exception("No auth details provided for Github. Define GITHUB_TOKEN.")
+
+        session.headers.update(
+            {
+                "Authorization": f"Bearer {token}",
+                "Accept": "application/vnd.github.v3+json",
+            }
+        )
+
     return session
 
 
+def get(
+    url: str, params: Optional[Dict[str, str]] = None, auth_required: bool = True
+) -> Dict[str, Any]:
+    """
+    Issue a GET request to the GitHub API
+
+    Args:
+        url (str): Github API URL
+        params (dict): Additional query parameters
+        auth_required (bool): Whether authentication should be required for the session
+
+    Returns:
+        Dict[str, Any]: GitHub response
+    """
+    session = _get_session(auth_required=auth_required)
+    LOGGER.debug(f"GET GitHub request url: {url}")
+    LOGGER.debug(f"GET GitHub request params: {params}")
+    resp = session.get(url, params=params)
+
+    try:
+        resp.raise_for_status()
+    except requests.HTTPError:
```

Comment:
Just wanna make sure. Does this also cover the MaxRetry exception?

Reply:
It doesn't. That error can be raised by `session.get` so will not necessarily be caught by this exception handler anyway. Prior to this change, retries were still enabled but the delay between attempts was 0. Therefore, this shouldn't change the current behavior for retry exception handling.

Reply:
I raised this question because from the recent experience when the retries session reaches a maximum attempt it just raises the MaxRetry exception with no details about what went wrong.

It would be great to log HTTP errors with an error code also for Retry exceptions. If you don't have the capacity to do it now can you please file a ticket? Thanks

Reply:
I think I found a nice solution to this. Setting `raise_on_status=False` on the `Retry` object will stop `MaxRetryError` exceptions from being raised by urllib3 (at least when there is a response code). `raise_for_status` would then raise them normally.

Reply:
Please take another look when you get a chance. Thanks!